src/DetailsView/components/save-assessment-button.tsx

```diff
@@ -1,29 +1,86 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { Checkbox, Dialog, DialogFooter, DialogType, PrimaryButton, Stack } from '@fluentui/react';
+import { useBoolean } from '@fluentui/react-hooks';
 import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
+import { UserConfigMessageCreator } from 'common/message-creators/user-config-message-creator';
+import { NamedFC } from 'common/react/named-fc';
+import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
 import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
+import styles from 'DetailsView/components/common-dialog-styles.scss';
 import * as React from 'react';
 
 export interface SaveAssessmentButtonDeps {
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
+    userConfigMessageCreator: UserConfigMessageCreator;
 }
 export interface SaveAssessmentButtonProps {
     download: string;
     href: string;
     deps: SaveAssessmentButtonDeps;
+    userConfigurationStoreData: UserConfigurationStoreData;
 }
 
-export class SaveAssessmentButton extends React.Component<SaveAssessmentButtonProps> {
-    public render(): JSX.Element {
+export const SaveAssessmentButton = NamedFC<SaveAssessmentButtonProps>(
+    'SaveAssessmentButton',
+    props => {
+        const [hideDialog, { toggle: toggleHideDialog }] = useBoolean(true);
```

Comment:
I haven't seen these fluent-UI hook utility functions before! I understand it now but had to trace it at first. I think(?) this is the first usage of fluent-ui/react-hooks so we might consider a quick ping to interested web-folks to validate that there are no concerns. I'm good either way 🙂